src/main/java/core/basesyntax/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void processOperation(FruitTransaction fruitTransaction) {
+        String fruit = fruitTransaction.getFruit();
+        int amountInStore = FruitStorage.fruitRepository.get(fruit);
+        int amountToBuy = fruitTransaction.getQuantity();
+        if (amountInStore < amountToBuy) {

```

Comment:
According to the checklist, you should check the result balance before saving it in the Storage - it should be positive. After line 15, you need to ensure that the resulting amount is not negative before putting it back into the repository.